### PR TITLE
fib-mem.lua

### PR DESCRIPTION
**Pull request by shakna-israel**
A memoized Lua, to go hand-in-hand with #30.

### CODE REVIEW

**Comment by drujensen** [comment]
Can you add a line in the optimized section in the readme with the recommended way to run it?

**Comment by shakna-israel** [comment]
I can, but can't fill in the `time` section, as I don't have the same hardware.

File: fib-mem.lua [hunk 1]
```diff
@@ -0,0 +1,17 @@
+local nums = {}
+local fib
+
+fib = function(n)
+  if n <= 1 then
+    return 1
+  else
```
Comment (drujensen) [inline fib-mem.lua]: can you remove the `else` since return should exit the function

File: fib-mem.lua [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    if nums[n] then
+      return nums[n]
+    else
```
Comment (drujensen) [inline fib-mem.lua]: also here, remove the else and just end

**Comment by shakna-israel** [comment]
Regarding changes...

```
local nums = {}
local fib

fib = function(n)
  if n <= 1 then return 1 end
  if nums[n] then return nums[n] end
  nums[n] = fib(n - 1) + fib(n - 2)
  return nums[n]
end

print(fib(46))
```

This drops the two elses you've asked for... But I am not sure why you want that.

It shouldn't have any real effect on the generated bytecode performance. They do differ by about 20bytes, but the opcodes are short-circuiting.

**Comment by drujensen** [comment]
Thanks, all the other examples do this and I want all the examples to be as consistent as possible.

**Pull request merged**